### ISSUE TITLE
Average T00 is nan

### ISSUE THREAD

**Issue by Lagrang3** [GrGadget/gevolution-1.2#6]
```
$ cat job.out 

  _   _      _         __ ,  _
 (_| (-' \/ (_) (_ (_| (  ( (_) /\/	version 1.2         running on 4 cores.
  -'

Version date: Mon Aug 16 11:39:17 2021 +0200
Commit: 216a82854a33d7cc5ae3bd415a30ed4ea1f8e8da

 initializing...
 baryon treatment set to: blend
 vector method set to: elliptic
 gravity theory set to: General Relativity
 cosmological parameters are: Omega_m0 = 0.3144, Omega_rad0 = 9.23088e-05, h = 0.673212
 parsing of settings file completed. 52 parameters found, 43 were used.
Initialization of the particles: part_simple
particles have individual mass
 262144 cdm particles initialized: maximum displacement = 0.026238 lattice units.
 initialization complete.

 cycle 0, background information: z = 19, average T00 = 0, background model = 0.3144
 cycle 0, time integration information: max |v| = 0.000669316 (cdm Courant factor = 0.00102112), time step / Hubble time = 0.02
 writing power spectra at z = 18.6059 (cycle 1), tau/boxsize = 2.23866
 writing power spectra at z = 15.7204 (cycle 9), tau/boxsize = 2.43819
 cycle 10, background information: z = 15.391, average T00 = -nan, background model = 0.3144

```

**Comment by Lagrang3** [comment]
In fact before the first iteration `T00 = 0` when the original code gives ` T00 = 0.3144`

**Comment by Lagrang3** [comment]
Did you check if particles has mass?

**Comment by Lagrang3** [comment]
Now I get ` T00 = 0.3144` before the first iteration. But I still get nans afterwards.

**Comment by Lagrang3** [comment]
Maybe particles are not being updated correctly. In fact the are being update with the old fields.

**Comment by Lagrang3** [comment]
I see sum over Phi gives nan at the first iteration even after potentials have been updated.

**Comment by Lagrang3** [comment]
solved after commit c33140d
